modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -153,6 +154,9 @@
     private final long IGNITE_EXCHANGE_MERGE_DELAY =
         IgniteSystemProperties.getLong(IgniteSystemProperties.IGNITE_EXCHANGE_MERGE_DELAY, 0);
 
+    /** */
+    private final int DIAGNOSTIC_WARN_LIMIT = IgniteSystemProperties.getInteger(IGNITE_DIAGNOSTIC_WARN_LIMIT, 10);
```

Comment:
Private final fields should not be in ALL_CAPS and should go after private static final. My suggestion is to compute it in dumpDebugInfo() every time. We are trying to avoid static fields since they have side-effects when running a lot of tests (can't change mid-flight).